lm_eval/tasks/score/utils.py

```diff
@@ -0,0 +1,228 @@
+        return result
+    return doc.map(process_batch, batched=True)
+
+def option_order_robustness_process_docs(doc: Dataset, 
+                                         template_file_path: str,
+                                         templates_key: str,
+                                         labels: list,
+                                         dataset_specific_preprocess: callable=None) -> Dataset:
+    try:
+        with open(template_file_path) as f:
+            prompt_template = json.load(f)[templates_key] 
+            prompt = prompt_template["prompt"]
+            options_format = prompt_template["options_format"]
+    except FileNotFoundError:
+        eval_logger.error("Prompt templates not found")
+        sys.exit()
+    
+    if dataset_specific_preprocess is not None:
+        doc = dataset_specific_preprocess(doc)
+
+    def repeat_doc_swap_correct_answer(batched_docs):
+
+        initial_len = len(next(iter(batched_docs.values())))
+        keys = list(batched_docs.keys())
+        new_batched_docs = {key: [] for key in keys}
+        new_batched_docs["always_same_option"] = []
+        new_batched_docs["prompt"] = []
+        new_batched_docs["options_format"] = []
+        new_batched_docs["original_answer_index"] = []
+        
+        for doc_ind in range(initial_len):
+            for label_ind, label in enumerate(labels):
+                new_batched_docs["original_answer_index"].append(batched_docs["answer_index"][doc_ind])
+                for key in keys:
+                    new_batched_docs[key].append(copy.deepcopy(batched_docs[key][doc_ind]))
+                    if label_ind < len(batched_docs["options"][doc_ind]):
+                        if key == "options":
+                            # Swap correct answer with label_ind option
+                            new_batched_docs[key][-1][label_ind] = batched_docs["options"][doc_ind][batched_docs["answer_index"][doc_ind]]
+                            new_batched_docs[key][-1][batched_docs["answer_index"][doc_ind]] = batched_docs["options"][doc_ind][label_ind]
+                        
+                        if key == "answer_index":
+                            new_batched_docs[key][-1] = label_ind
+
+                        if key == "answer":
+                            new_batched_docs[key][-1] = label
+
+                new_batched_docs["always_same_option"].append(label)
+                new_batched_docs["prompt"].append(prompt)
+                new_batched_docs["options_format"].append(options_format)
+        return new_batched_docs
+    return doc.map(repeat_doc_swap_correct_answer, batched=True)
+
+def robustness_doc_to_text(doc: Dataset) -> str:
+    upper_case = string.ascii_uppercase
+    lower_case = string.ascii_lowercase
+    prompt = doc["prompt"]
+    options_format = doc.get("options_format", "")
+    question = doc["question"]
+    catrgory = doc.get("category", "")
+    options = None
+    if options_format:
```

Comment:
these should also be removed right?

Reply:
I kept the logic of passing "prompt" and "options_format" separately from json, It is a more generic way of processing the documents I think. So in current setting this should not be removed.